### ISSUE TITLE
Size of Iden3Mobile.framework for iOS is too big (134,2 MB)

### ISSUE THREAD

**Issue by rauljareno** [iden3/iden3-mobile#125]
As an integrator wanting to use our technology, having to include a go-mobile library of 134 MB to the app the integrator is developing is too much. The size of this library should be reduced as much as possible.

The average app file size is now 38MB for iOS
https://sweetpricing.com/blog/2017/02/average-app-file-size/

**Comment by arnaubennassar** [comment]
Totally agree on that size matters. But is the framework size mapped 1:1 to the compiled app? Could you check that?

I think it will be quite hard to reduce the size of this artifact, keep in mind that this doesn't any kind of image or other heavy sized element, and the code it self shouldn't weight that much. On the other hand 3rd party libraries may be heavy but it's hard to modify them...

A part from that, the iOS release it's about 3 times bigger than the Android one, but I guess that it's still a lot.

**Comment by ed255** [comment]
134 MB indeed seems too big!  I have checked the library binary and it targets 4 architectures: [armv7] [i386] [x86_64] [arm64].  Considering that we are targeting iPhones and iPads, and that for some time iOS devices have been using arm64 (plus performance in 32 bits arch will be very poor), the app should only take the arm64 part, so the final size should be divided by 4 (33.5 MB).

I've also seen this related thread https://github.com/golang/go/issues/15223 It states that after doing static linking with the library, the final app size is much smaller than the framework size.

**Comment by rauljareno** [comment]
Cool, we will need then to create too a demo app for the iOS Mobile SDK (wrapper) which will only include the core library framework to see the final app size is also reduces, so we can check this properly. Thanks @ed255 and @arnaubennassar

**Comment by arnaubennassar** [comment]
just by using `--symlinks` when zipping the framework reduced it's size to 41.5MB: https://github.com/iden3/iden3-mobile/releases/tag/v0.0.2

Do you think that this is good enough to close this issue @rauljareno  @ed255 ?

**Comment by rauljareno** [comment]
@arnaubennassar after unzipping the 41 MB zip file, the framework size is 135 MB still, so is the same problem.

**Comment by arnaubennassar** [comment]
true, in that case we'll have to wait until we see the real weight on a app

**Comment by rauljareno** [comment]
@arnaubennassar I would create two tasks from this issue, one is to check how to include only the architecture arm64 of the framework to reduce its size as @ed255 was mentioning, and the other one to create a demo app for iOS to check if the size is also more reduced.
